pubgrep/pubgrep.py

```diff
@@ -0,0 +1,520 @@
+    return hlgap
+
+
+def get_charge_from_xtb_output(xtb_out: str, verbosity: int) -> int:
+    """
+    This function is used to extract the total charge from the xtb output.
+    """
+    # load fourth entry of a line with ":: total charge" of xtb.out into a variable
+    chrg = 0
+    for line in xtb_out.split("\n"):
+        if ":: total charge" in line:
+            chrg = round(float(line.split()[3]))
+            break
+    if verbosity > 1:
+        print(" " * 3 + f"Total charge: {chrg:6d}")
+    return chrg
+
+
+### Argument parsing functions
+def get_args() -> argparse.Namespace:
+    """
+    This function is used to parse the command line arguments.
+    """
+    parser = argparse.ArgumentParser(
+        description=header()
+        + "\nThis program searches CIDs from the PubChem Database based on a list of compounds.",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+    parser.add_argument(
+        "compounds",
+        help="List of compounds to search for.",
+        nargs="+",
+        type=str,
+        default=None,
+        action="store",
+    )
+    parser.add_argument(
+        "--input",
+        default="name",
+        choices=["name", "cid", "smiles", "cas", "inchi"],
+        help="Input format",
+    )
+    parser.add_argument(
+        "--output", default="sdf", choices=["sdf", "logP", "list"], help="Output format"
+    )
+    parser.add_argument(
+        "--skip", action="store_true", help="Skip PubChem server testing"
+    )
+    parser.add_argument(
+        "--verbosity",
+        type=int,
+        default=0,
+        choices=[0, 1, 2, 3],
+        help="Verbosity level",
+        required=False,
+    )
+    parser.add_argument(
+        "--opt",
+        action="store_true",
+        help="Optimize the structure of the compound",
+        required=False,
+        default=False,
+    )
+    parser.add_argument(
+        "--basedir",
+        type=Path,
+        default="pubchem_compounds",
+        help="Base directory for the compound data",
+        required=False,
+    )
+    parser.add_argument(
+        "--hlgap_thr",
+        type=float,
+        default=0.5,
+        help="Threshold for the HOMO-LUMO gap",
+        required=False,
+    )
+    return parser.parse_args()
+
+
+def cli():
+    """
+    This function is used to parse the command line arguments.
+    """
+    args = get_args()
+    verbosity = args.verbosity
+    compounds = args.compounds
+    input_format = args.input
+    output_format = args.output
+    optimization = args.opt
+    skip = args.skip
+    basedir = args.basedir
+    hlgap_thr = args.hlgap_thr
+
+    xtb_path = Path(shutil.which("xtb")).resolve()
+
+    pubgrep(
+        xtb_path=xtb_path,
+        basedir=basedir,
+        compounds=compounds,
+        input_format=input_format,
+        output_format=output_format,
+        optimization=optimization,
+        hlgap_thr=hlgap_thr,
+        skip=skip,
+        verbosity=verbosity,
+    )
```

Comment:
Consider returning an exit code here, i.e., `return 0` or `return 1`. This will be shown in `$?` (bash).

Reply:
Do you mean something like this:

https://github.com/grimme-lab/PubGrep/pull/1/commits/7ce6f19599f2d29116b001f8542407bfc0b3ebf0

Reply:
No, you really need to just return an integer. This value is propagated by `raise SystemExit`.

Reply:
And again, not a big fan of generic exceptions. This overrides the exit codes of all errors (e.g. also KeyboardInterrupt).

Reply:
Hope I got what you meant. Should be fixed in the latest commit.